### DISCUSSION TITLE
How to make REST API-only app (and 404's and other errors return JSON always)?

### DISCUSSION THREAD

**Discussion by lancejpollard** [vercel/next.js#69709] [App Router]
How can I make it so there is absolutely zero HTML returned on a Vercel hosted Next.js app? Including handling `not-found`, `error` and `global-error` responses which never return HTML but only return pure JSON?

**Comment by Juneezee** [comment]
> so there is absolutely zero HTML

Next.js is a React framework and isn't really designed for a pure API backend. Have you considered options such as 

1. [nestjs/nest](https://github.com/nestjs/nest)
2. [expressjs/express](https://github.com/expressjs/express)
3. [fastify/fastify](https://github.com/fastify/fastify)
